cmd/frontend/internal/frontend/service.go

```diff
@@ -309,6 +317,17 @@ func (fes *FrontEndService) Monitor(ctx context.Context) error {
 						if err := announceFrontend(fes.targetRegistryClient); err != nil {
 							logrus.Infof("FrontEndService: failed to announce frontend connectivity (err: %v)", err)
 						}
+						// refresh NSP entry
+						var refreshCtx context.Context
+						refreshCtx, refreshCancel = context.WithCancel(context.TODO())
+						defer refreshCancel()
```

Comment:
What happens on link flapping? Imho this leads to memory leak:
Every time when the external connectivity recovers a new deferred function is added to the stack.

I would sg like this to the end of the for loop instead:
```
if refreshCancel != nil {
	refreshCancel()
}
```